### ISSUE TITLE
WakaTime Error in Visual Studio Code - Windows 10

### ISSUE THREAD

**Issue by xinthose** [wakatime/vscode-wakatime#228]
**Visual Studio Debug**
```
Search "[WakaTime]" (13 hits in 1 file of 1 searched)
  C:\Users\ADunsmoor\Downloads\vscode-app-1630504212594.log (13 hits)
	Line 331: console.ts:137 [Extension Host] [WakaTime][DEBUG] Initializing WakaTime v17.0.0
	Line 377: console.ts:137 [Extension Host] [WakaTime][DEBUG] Current wakatime-cli version is v1.22.0
	Line 378: console.ts:137 [Extension Host] [WakaTime][DEBUG] Checking for updates to wakatime-cli...
	Line 380: console.ts:137 [Extension Host] [WakaTime][DEBUG] GitHub API Response 304
	Line 381: console.ts:137 [Extension Host] [WakaTime][DEBUG] wakatime-cli is up to date
	Line 382: console.ts:137 [Extension Host] [WakaTime][DEBUG] WakaTime initialized.
	Line 383: console.ts:137 [Extension Host] [WakaTime][DEBUG] Status bar icon enabled.
	Line 384: console.ts:137 [Extension Host] [WakaTime][DEBUG] Fetching coding activity for Today from api: C:\Users\ADunsmoor\.wakatime\wakatime-cli-windows-amd64.exe --today --plugin "\"vscode/1.59.1 vscode-wakatime/17.0.0\"" --config C:\Users\ADunsmoor\.wakatime.cfg --logfile C:\Users\ADunsmoor\.wakatime.log
	Line 392: console.ts:137 [Extension Host] [WakaTime][ERROR] Error: Command failed: C:\Users\ADunsmoor\.wakatime\wakatime-cli-windows-amd64.exe --today --plugin "vscode/1.59.1 vscode-wakatime/17.0.0" --config C:\Users\ADunsmoor\.wakatime.cfg --logfile C:\Users\ADunsmoor\.wakatime.log
	Line 413: console.ts:137 [Extension Host] [WakaTime][DEBUG] Error fetching today coding activity (1); Check your C:\Users\ADunsmoor\.wakatime.log file for more details
	Line 421: console.ts:137 [Extension Host] [WakaTime][DEBUG] Sending heartbeat: C:\Users\ADunsmoor\.wakatime\wakatime-cli-windows-amd64.exe --entity c:\Users\ADunsmoor\Documents\GitHub\EVC\8300\EVC8365\client\src\app\please-wait\please-wait.component.ts --plugin "\"vscode/1.59.1 vscode-wakatime/17.0.0\"" --alternate-project EVC8365 --config C:\Users\ADunsmoor\.wakatime.cfg --log-file C:\Users\ADunsmoor\.wakatime.log
	Line 422: console.ts:137 [Extension Host] [WakaTime][ERROR] Error: Command failed: C:\Users\ADunsmoor\.wakatime\wakatime-cli-windows-amd64.exe --entity c:\Users\ADunsmoor\Documents\GitHub\EVC\8300\EVC8365\client\src\app\please-wait\please-wait.component.ts --plugin "vscode/1.59.1 vscode-wakatime/17.0.0" --alternate-project EVC8365 --config C:\Users\ADunsmoor\.wakatime.cfg --log-file C:\Users\ADunsmoor\.wakatime.log
	Line 443: console.ts:137 [Extension Host] [WakaTime][ERROR] Unknown Error (1); Check your C:\Users\ADunsmoor\.wakatime.log file for more details
```
**.wakatime.log**
```
{"caller":"/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:58","func":"Run","level":"debug","message":"command: version","now":"2021-09-01T08:54:08-05:00","version":"v1.22.0"}
{"caller":"/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:76","func":"Run","level":"debug","message":"command: today","now":"2021-09-01T08:54:09-05:00","version":"v1.22.0"}
{"caller":"/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:88","func":"Run","level":"debug","message":"command: heartbeat","now":"2021-09-01T08:54:48-05:00","version":"v1.22.0"}
{"caller":"/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/heartbeat/heartbeat.go:85","func":"SendHeartbeats","level":"debug","message":"heartbeat params: category: 'coding', cursor position: '', entity: 'c:\\Users\\ADunsmoor\\Documents\\GitHub\\EVC\\8300\\EVC8365\\client\\src\\app\\please-wait\\please-wait.component.ts', entity type: 'file', num extra heartbeats: 0, is write: false, language: '', line number: '', lines in file: '', offline disabled: false, offline queue file: '', offline sync max: 1000, time: 1630504488.35987, api params: (api key: '77db...', api url: 'https://api.wakatime.com/api/v1', backoff at: '', backoff retries: 0, hostname: 'PC123', plugin: 'vscode/1.59.1 vscode-wakatime/17.0.0', timeout: 2m0s, disable ssl verify: false, proxy url: '', ssl cert filepath: ''), filter params: (exclude: '[]', exclude unknown project: false, include: '[]', include only with project file: false), project params: (alternate: 'EVC8365', disable submodule: '[]', map patterns: '[]', override: ''), sanitize params: (hide branch names: '[]', hide file names: '[]', hide project names: '[]')","now":"2021-09-01T08:54:48-05:00","version":"v1.22.0"}
```
Uninstalling and re-installing the plugin does not help.  My WakaTime API key is set.  A linux laptop on the same network is working fine with VS Code and WakaTime using the same account.  This randomly happened to me on Windows.  I have been using it fine without issues for months.

**Comment by alanhamlett** [comment]
Try turning off debug mode since this is a normal error that's usually hidden.

**Comment by xinthose** [comment]
@alanhamlett I did that.  I see this error a lot from VS Code:

`console.ts:137 [Extension Host] [WakaTime][ERROR] panic: exec: "cmd": executable file not found in %PATH% [recovered]	panic: exec: "cmd": executable file not found in %PATH%goroutine 1 [running]:github.com/matishsiao/goInfo.GetInfo(0xc0003b57e0)	/Users/runner/go/pkg/mod/github.com/wakatime/go!info@v0.1.0-wakatime.2/goInfo_windows.go:20 +0x4a5github.com/wakatime/wakatime-cli/pkg/heartbeat.UserAgent(0xc000018391, 0x24, 0xc000594fe0, 0x13fb9d4)	/Users/runner/work/wakatime-cli/wakatime-cli/pkg/heartbeat/heartbeat.go:170 +0x3bgithub.com/wakatime/wakatime-cli/pkg/api.WithUserAgent(0xc000018391, 0x24, 0x2)	/Users/runner/work/wakatime-cli/wakatime-cli/pkg/api/option.go:171 +0x45github.com/wakatime/wakatime-cli/cmd/legacy/legacyapi.newClient(0x0, 0x0, 0x0, 0x0, 0x0, 0xc0004c5aa8, 0x5, 0xc0000d6211, 0x24, 0xc000018391, ...)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/legacyapi/api.go:74 +0x197github.com/wakatime/wakatime-cli/cmd/legacy/legacyapi.NewClientWithoutAuth(...)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/legacyapi/api.go:27github.com/wakatime/wakatime-cli/cmd/legacy.sendDiagnostics(0xc000366780, 0x0, 0x0, 0xc00046ac00, 0xbff)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:214 +0x1efgithub.com/wakatime/wakatime-cli/cmd/legacy.runCmd.func1(0xc0000c01e0, 0x0, 0xc000366780, 0xc00010f3b0)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:184 +0x115panic(0x16997a0, 0xc00064b400)	/Users/runner/hostedtoolcache/go/1.16.7/x64/src/runtime/panic.go:965 +0x1c7github.com/matishsiao/goInfo.GetInfo(0xfc6fd81eb3d377d9)	/Users/runner/go/pkg/mod/github.com/wakatime/go!info@v0.1.0-wakatime.2/goInfo_windows.go:20 +0x4a5github.com/wakatime/wakatime-cli/pkg/heartbeat.UserAgent(0xc000018391, 0x24, 0xc00061f6b8, 0x13fb9d4)	/Users/runner/work/wakatime-cli/wakatime-cli/pkg/heartbeat/heartbeat.go:170 +0x3bgithub.com/wakatime/wakatime-cli/pkg/api.WithUserAgent(0xc000018391, 0x24, 0x0)	/Users/runner/work/wakatime-cli/wakatime-cli/pkg/api/option.go:171 +0x45github.com/wakatime/wakatime-cli/cmd/legacy/legacyapi.newClient(0x0, 0x0, 0x0, 0x0, 0x0, 0xc000022528, 0x5, 0xc0000d6211, 0x24, 0xc000018391, ...)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/legacyapi/api.go:74 +0x197github.com/wakatime/wakatime-cli/cmd/legacy/legacyapi.NewClient(0x0, 0x0, 0x0, 0x0, 0x0, 0xc000022528, 0x5, 0xc0000d6211, 0x24, 0xc000018391, ...)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/legacyapi/api.go:21 +0x168github.com/wakatime/wakatime-cli/cmd/legacy/today.Today(0xc000366780, 0x18, 0x16aaa20, 0x242e6fdb401, 0xc0000c01e0)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/today/today.go:56 +0x1b8github.com/wakatime/wakatime-cli/cmd/legacy/today.Run(0xc000366780, 0xc00010f3b0, 0xc0000c01e0, 0xc00068b4f0)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/today/today.go:19 +0x47github.com/wakatime/wakatime-cli/cmd/legacy.runCmd(0xc000366780, 0xc00068b400, 0x17c54d8, 0x0)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:192 +0xe4github.com/wakatime/wakatime-cli/cmd/legacy.RunCmd(0xc000366780, 0x0, 0x17c54d8)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:156 +0x45github.com/wakatime/wakatime-cli/cmd/legacy.Run(0xc0001342c0, 0xc000366780)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:78 +0x76cgithub.com/wakatime/wakatime-cli/cmd.NewRootCMD.func1(0xc0001342c0, 0xc00015c070, 0x0, 0x7)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/root.go:28 +0x3bgithub.com/spf13/cobra.(*Command).execute(0xc0001342c0, 0xc000126010, 0x7, 0x7, 0xc0001342c0, 0xc000126010)	/Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:854 +0x2c2github.com/spf13/cobra.(*Command).ExecuteC(0xc0001342c0, 0xc00004ff78, 0xc00061ff78, 0xf7670b)	/Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375github.com/spf13/cobra.(*Command).Execute(...)	/Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895github.com/wakatime/wakatime-cli/cmd.Execute()	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/root.go:224 +0x32main.main()	/Users/runner/work/wakatime-cli/wakatime-cli/main.go:6 +0x27`

also this error:

`console.ts:137 [Extension Host] [WakaTime][ERROR] Error: Command failed: C:\Users\ADunsmoor\.wakatime\wakatime-cli-windows-amd64.exe --today --plugin "vscode/1.59.1 vscode-wakatime/17.0.0" --config C:\Users\ADunsmoor\.wakatime.cfg --logfile C:\Users\ADunsmoor\.wakatime.logpanic: exec: "cmd": executable file not found in %PATH% [recovered]	panic: exec: "cmd": executable file not found in %PATH%goroutine 1 [running]:github.com/matishsiao/goInfo.GetInfo(0xc0003b57e0)	/Users/runner/go/pkg/mod/github.com/wakatime/go!info@v0.1.0-wakatime.2/goInfo_windows.go:20 +0x4a5github.com/wakatime/wakatime-cli/pkg/heartbeat.UserAgent(0xc000018391, 0x24, 0xc000594fe0, 0x13fb9d4)	/Users/runner/work/wakatime-cli/wakatime-cli/pkg/heartbeat/heartbeat.go:170 +0x3bgithub.com/wakatime/wakatime-cli/pkg/api.WithUserAgent(0xc000018391, 0x24, 0x2)	/Users/runner/work/wakatime-cli/wakatime-cli/pkg/api/option.go:171 +0x45github.com/wakatime/wakatime-cli/cmd/legacy/legacyapi.newClient(0x0, 0x0, 0x0, 0x0, 0x0, 0xc0004c5aa8, 0x5, 0xc0000d6211, 0x24, 0xc000018391, ...)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/legacyapi/api.go:74 +0x197github.com/wakatime/wakatime-cli/cmd/legacy/legacyapi.NewClientWithoutAuth(...)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/legacyapi/api.go:27github.com/wakatime/wakatime-cli/cmd/legacy.sendDiagnostics(0xc000366780, 0x0, 0x0, 0xc00046ac00, 0xbff)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:214 +0x1efgithub.com/wakatime/wakatime-cli/cmd/legacy.runCmd.func1(0xc0000c01e0, 0x0, 0xc000366780, 0xc00010f3b0)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:184 +0x115panic(0x16997a0, 0xc00064b400)	/Users/runner/hostedtoolcache/go/1.16.7/x64/src/runtime/panic.go:965 +0x1c7github.com/matishsiao/goInfo.GetInfo(0xfc6fd81eb3d377d9)	/Users/runner/go/pkg/mod/github.com/wakatime/go!info@v0.1.0-wakatime.2/goInfo_windows.go:20 +0x4a5github.com/wakatime/wakatime-cli/pkg/heartbeat.UserAgent(0xc000018391, 0x24, 0xc00061f6b8, 0x13fb9d4)	/Users/runner/work/wakatime-cli/wakatime-cli/pkg/heartbeat/heartbeat.go:170 +0x3bgithub.com/wakatime/wakatime-cli/pkg/api.WithUserAgent(0xc000018391, 0x24, 0x0)	/Users/runner/work/wakatime-cli/wakatime-cli/pkg/api/option.go:171 +0x45github.com/wakatime/wakatime-cli/cmd/legacy/legacyapi.newClient(0x0, 0x0, 0x0, 0x0, 0x0, 0xc000022528, 0x5, 0xc0000d6211, 0x24, 0xc000018391, ...)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/legacyapi/api.go:74 +0x197github.com/wakatime/wakatime-cli/cmd/legacy/legacyapi.NewClient(0x0, 0x0, 0x0, 0x0, 0x0, 0xc000022528, 0x5, 0xc0000d6211, 0x24, 0xc000018391, ...)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/legacyapi/api.go:21 +0x168github.com/wakatime/wakatime-cli/cmd/legacy/today.Today(0xc000366780, 0x18, 0x16aaa20, 0x242e6fdb401, 0xc0000c01e0)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/today/today.go:56 +0x1b8github.com/wakatime/wakatime-cli/cmd/legacy/today.Run(0xc000366780, 0xc00010f3b0, 0xc0000c01e0, 0xc00068b4f0)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/today/today.go:19 +0x47github.com/wakatime/wakatime-cli/cmd/legacy.runCmd(0xc000366780, 0xc00068b400, 0x17c54d8, 0x0)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:192 +0xe4github.com/wakatime/wakatime-cli/cmd/legacy.RunCmd(0xc000366780, 0x0, 0x17c54d8)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:156 +0x45github.com/wakatime/wakatime-cli/cmd/legacy.Run(0xc0001342c0, 0xc000366780)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/legacy/run.go:78 +0x76cgithub.com/wakatime/wakatime-cli/cmd.NewRootCMD.func1(0xc0001342c0, 0xc00015c070, 0x0, 0x7)	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/root.go:28 +0x3bgithub.com/spf13/cobra.(*Command).execute(0xc0001342c0, 0xc000126010, 0x7, 0x7, 0xc0001342c0, 0xc000126010)	/Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:854 +0x2c2github.com/spf13/cobra.(*Command).ExecuteC(0xc0001342c0, 0xc00004ff78, 0xc00061ff78, 0xf7670b)	/Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375github.com/spf13/cobra.(*Command).Execute(...)	/Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895github.com/wakatime/wakatime-cli/cmd.Execute()	/Users/runner/work/wakatime-cli/wakatime-cli/cmd/root.go:224 +0x32main.main()	/Users/runner/work/wakatime-cli/wakatime-cli/main.go:6 +0x27`

`.wakatime.log` is empty now

**Comment by alanhamlett** [comment]
If you restart vscode this error should go away. 👍 

Thanks for reporting it, turns out a dependency we used called `panic` instead of returning an error like we had expected. Fixed in v1.22.1 of wakatime-cli.

**Comment by xinthose** [comment]
@alanhamlett Thank you!  It is working now for me.
